src/ExplorerServiceProvider.php

```diff
@@ -33,27 +34,7 @@ public function boot(): void
         }
 
         $this->app->bind(ElasticClientFactory::class, function () {
-            $client = ClientBuilder::create()->setHosts([config('explorer.connection')]);
-
-            if(config()->has('explorer.connection.api')) {
-                $client->setApiKey(
-                    config('explorer.connection.api.id'),
-                    config('explorer.connection.api.key')
-                );
-            }
-
-            if(config()->has('explorer.connection.auth')) {
-                $client->setBasicAuthentication(
-                    config('explorer.connection.auth.username'),
-                    config('explorer.connection.auth.password')
-                );
-            }
-
-            if(config()->has('explorer.connection.ssl.verify')) {
-                $client->setSSLVerification(config('explorer.connection.ssl.verify'));
-            }
-
-            return new ElasticClientFactory($client->build());
+            return ElasticClientBuilder::fromConfig(config())->build();
```

Comment:
Now you are passing the entire Laravel config, that's not really what I'd want. I suppose `config('explorer')` works as well.

Reply:
Ah, you are actually using the config class I see...

Reply:
If you don't want to work with arrays, then not really 🤔